### PR TITLE
CS Search: Experimental boosting & name resolution

### PR DESCRIPTION
**Pull request by urosj**
To test, delete existing index and rerun essync with new entity.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/446/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/447/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/448/

File: internal/charmstore/search.go [hunk 1]
```diff
@@ -126,7 +126,14 @@ func (store *Store) Search(sp SearchParams) (SearchResult, error) {
...
+		"CharmMeta.Categories": 5,
+		// todo (mhilton) tests fail if BundleData.Tags are enabled
+		//"BundleData.Tags":         5,
+		"CharmProvidedInterfaces": 3,
+		"CharmRequiredInterfaces": 3,
+		"CharmMeta.Description":   1,
+		"BundleReadMe":            1,
 	}
 	if sp.AutoComplete {
 		fields["CharmMeta.Name.ngrams"] = 10
```
Comment (mhilton) [inline internal/charmstore/search.go:139]: You might want to reconsider these weightings too.

**Comment by mhilton** [comment]
This seems to work better at finding things than before. :+1: with QA.

**Comment by urosj** [comment]
:shipit:

**Pull request merged**